### ISSUE TITLE
fatal: ref HEAD is not symbolic ref (on release published event)

### ISSUE THREAD

**Issue by ldelossa** [peter-evans/create-pull-request#522]
### Subject of the issue

I have a github action trigged by a release publish.
I'm trying to use this action to create a PR from this event.
I get the following error:

```
Checking the checked out ref
  /usr/bin/git symbolic-ref HEAD --short
  fatal: ref HEAD is not a symbolic ref
  ##[error]The checked out ref is not a valid base for a pull request. Unable to continue.
```

### Steps to reproduce

You can see my workflow file here:
https://github.com/ldelossa/gcl/blob/master/.github/workflows/release.yml

and you can see the encountered error here:
https://github.com/ldelossa/gcl/runs/1017534178?check_suite_focus=true

I have tried one work around I found on the issue tracker, but the problem seems to persist.

**Comment by zlesnr** [comment]
I'm bumping into this also, and will try the above solution tomorrow.  Though it does make me wonder, why isn't a tag a valid base for a PR?  The tag should just point to the sha of the commit, which should be a valid base, correct?

**Comment by zlesnr** [comment]
I'm also trying to use this to update a changelog on release.

**Comment by ldelossa** [comment]
I think when you check out a tag HEAD is no longer a valid reference.

**Comment by peter-evans** [comment]
@zlesnr Checking out a commit is not the same as checking out a `ref`. The former will cause a "Detached HEAD" state where there is no `ref` associated with the checked out commit. This state is not a valid base because pull requests work using `refs`, e.g. `my-branch` -> `master`.

If the above solution doesn't solve your use case, please post here for help.

**Comment by zlesnr** [comment]
I see, thank you for the explanation.  In the case above, and as long as the master branch contains the sha that the tag points at, then the initial checkout could be to the master branch, correct?  Then skip the upload/download of the artifact before creating the PR.

I was thinking that this action would do something like `git checkout -b myupdatebranch v2.2.2` to base the PR on the commit, but I suppose that all the checkout work is being handled from the checkout action.  Is that correct?

**Comment by peter-evans** [comment]
@zlesnr In [this workflow](https://github.com/peter-evans/create-pull-request/issues/522#issuecomment-678732429) you could replace the initial checkout with the following, and skip upload/download:

```
      - uses: actions/checkout@v2
        with:
          ref: master
          fetch-depth: 0
```
While this would work fine, it is not guaranteed to checkout the tagged commit. There is a chance that in between the tag being pushed and checking out the master branch in the workflow, another commit is made to master. That would pollute your changelog with a commit that wasn't in the release. If you think that scenario is unlikely to happen in your case, then it might be ok.

> I was thinking that this action would do something like `git checkout -b myupdatebranch v2.2.2` to base the PR on the commit, but I suppose that all the checkout work is being handled from the checkout action. Is that correct?

The action does checkout a temporary branch for the changes, and can even rebase the changes on a different base to the checked out one. Now that you point it out, I might be able to create a temporary branch for cases where `ref HEAD is not a symbolic ref` and rebase onto the `base` input. I will investigate this and let you know if I get it working.

**Comment by peter-evans** [comment]
@ldelossa @zlesnr Thank you both for motivating me to take a closer look at this. I realised it is possible for the action to support being checked out on a commit, not a branch, provided the `base` input is supplied.

I've released `v3.2.0` with support for this. Please see [events which checkout a commit](https://github.com/peter-evans/create-pull-request/blob/master/docs/concepts-guidelines.md#events-which-checkout-a-commit)

@ldelossa I think you can now simplify your workflow to the following. Note that I added `rm git-chglog` to make sure it's not included in the pull request.

```yml
name: Release Actions

on:
  release:
    types: [published]

jobs:
  changelog-bump-pr:
    runs-on: 'ubuntu-latest'
    steps:
      - name: Checkout the tag
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: ChangeLogBump
        shell: bash
        run: |
          curl -o git-chglog -L https://github.com/git-chglog/git-chglog/releases/download/0.9.1/git-chglog_linux_amd64
          chmod u+x git-chglog
          ./git-chglog -o CHANGELOG.md
          rm git-chglog
      - name: PullRequest
        uses: peter-evans/create-pull-request@v3.2.0
        with:
          title: "${{ github.ref }} Changelog Bump"
          body: "This is an automated change log bump following a release."
          commit-message: "cidi: ${{ github.ref }} changelog bump"
          branch: "${{github.ref}}-changelog-bump"
          base: master
```

**Comment by zlesnr** [comment]
Oh awesome @peter-evans.  I'll try that out this week.  Thanks for the support!
